util/patricia_trie/src/lib.rs

```diff
@@ -49,155 +49,159 @@ mod lookup;
 mod nibbleslice;
 mod nibblevec;
 
-pub use self::triedbmut::TrieDBMut;
 pub use self::triedb::{TrieDB, TrieDBIterator};
+pub use self::triedbmut::TrieDBMut;
 pub use self::sectriedbmut::SecTrieDBMut;
 pub use self::sectriedb::SecTrieDB;
 pub use self::fatdb::{FatDB, FatDBIterator};
 pub use self::fatdbmut::FatDBMut;
 pub use self::recorder::Recorder;
+use node_codec::NodeCodec;
+
+// REVIEW: is this overdoing it?
+pub type KeccakRlpNodeCodec = node_codec::RlpNodeCodec<hashdb::KeccakHasher>;
```

Comment:
it might be overdoing it a bit. I think the most idiomatic thing is to have `RlpNodeCodec` stuff defined in its own crate, and `KeccakHasher` defined in its own crate as well, without `patricia-trie` depending on either. Then there might be some facade crate (`patricia-trie-ethereum`) which re-exports the trie types and traits from the `patricia-trie` crate and depends exports some typedefs using `KeccakHasher` and `RlpNodeCodec`. IMO those typedefs should have simple names, that don't mention `Keccak` or  `Rlp` since they are apparent from context.

But that's if you want to go all out. What's here is enough for generalization in Polkadot.